x-pack/solutions/security/plugins/security_solution/server/assistant/tools/integration_knowledge/integration_knowledge_tool.ts

```diff
@@ -0,0 +1,134 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import { tool } from '@langchain/core/tools';
+
+import { z } from '@kbn/zod';
+import type { AssistantTool, AssistantToolParams } from '@kbn/elastic-assistant-plugin/server';
+import type { ContentReference } from '@kbn/elastic-assistant-common';
+import { contentReferenceString } from '@kbn/elastic-assistant-common';
+import {
+  hrefReference,
+  knowledgeBaseReference,
+} from '@kbn/elastic-assistant-common/impl/content_references/references';
+import { Document } from 'langchain/document';
+import type { Require } from '@kbn/elastic-assistant-plugin/server/types';
+import { APP_UI_ID } from '../../../../common';
+
+export type IntegrationKnowledgeToolParams = Require<AssistantToolParams, 'assistantContext'>;
+
+const INTEGRATIONS_BASE_PATH = '/app/integrations/detail';
+
+const toolDetails = {
+  // note: this description is overwritten when `getTool` is called
+  // local definitions exist ../elastic_assistant/server/lib/prompt/tool_prompts.ts
+  // local definitions can be overwritten by security-ai-prompt integration definitions
+  description:
+    'Call this for knowledge from Fleet-installed integrations, which contains information on how to configure and use integrations for data ingestion.',
+  id: 'integration-knowledge-tool',
+  name: 'IntegrationKnowledgeTool',
+};
+
+const INTEGRATION_KNOWLEDGE_INDEX = '.integration_knowledge';
+
+export const INTEGRATION_KNOWLEDGE_TOOL: AssistantTool = {
+  ...toolDetails,
+  sourceRegister: APP_UI_ID,
+  isSupported: (params: AssistantToolParams): params is IntegrationKnowledgeToolParams => {
+    const { assistantContext } = params;
+    return assistantContext != null;
+  },
```

Comment:
It might be good to add a check here for the existence of the `.integration_knowledge` index that way the tool isn't registered if there is no data:

<p align="center">
  <img width="700" src="https://github.com/user-attachments/assets/9984a109-d572-496c-97a7-27b601f50081" />
</p>

While we're trying to move to more of a 'fail within the tool' approach when possible to ensure a more predictable UX for the user, I think it makes sense to not register if the index isn't available as this will improve latency/churn from failed tool calls.

Reply:
interesting, I thought this would be already created based on ES work, but it seems only the mapping are set up. I'll add a check for this